src/components/JSONExportDialog.tsx

```diff
@@ -0,0 +1,135 @@
+import React, { useState } from "react";
+import { ActionsManagerInterface } from "../actions/types";
+import { NonDeletedExcalidrawElement } from "../element/types";
+import { t } from "../i18n";
+import { useIsMobile } from "./App";
+import { AppState } from "../types";
+import { Dialog } from "./Dialog";
+import { exportToFileIcon, link, save } from "./icons";
+import { ToolButton } from "./ToolButton";
+import { actionSaveAsScene } from "../actions/actionExport";
+import { Card } from "./Card";
+
+import "./ExportDialog.scss";
+
+export const ErrorCanvasPreview = () => {
+  return (
+    <div>
+      <h3>{t("canvasError.cannotShowPreview")}</h3>
+      <p>
+        <span>{t("canvasError.canvasTooBig")}</span>
+      </p>
+      <em>({t("canvasError.canvasTooBigTip")})</em>
+    </div>
+  );
+};
+
+export type ExportCB = (
+  elements: readonly NonDeletedExcalidrawElement[],
+  scale?: number,
+) => void;
+
+const JSONExportModal = ({
+  elements,
+  appState,
+  actionManager,
+  onExportToBackend,
+}: {
+  appState: AppState;
+  elements: readonly NonDeletedExcalidrawElement[];
+  actionManager: ActionsManagerInterface;
+  onExportToBackend?: ExportCB;
+  onCloseRequest: () => void;
+}) => {
+  const exportedElements = elements;
+
+  return (
+    <div className="ExportDialog">
+      <div
+        style={{
+          display: "grid",
+          gridTemplateColumns: "repeat(auto-fit, minmax(200px, 1fr))",
+          justifyItems: "center",
+        }}
+      >
+        <Card color="lime">
+          <div className="Card-icon">{exportToFileIcon}</div>
+          <h2>Save to disk</h2>
```

Comment:
Lets either use Export/ save  instead of both depending on the icon we finalize

Reply:
I purposefully used both not to repeat ourselves + it never hurts to use different terms so as to explain a concept more fully (and help people associate the two words together).